packages/xstate-inspect/src/browser.ts

```diff
@@ -138,19 +139,24 @@ export function inspect(
     inspectService.send({ type: 'unload' });
   });
 
+  const stringifyWithSerializer = (value: any) =>
+    stringify(value, options?.serialize);
+
   devTools.onRegister((service) => {
     inspectService.send({
       type: 'service.register',
-      machine: stringify(service.machine),
-      state: stringify(service.state || service.initialState),
+      machine: stringifyWithSerializer(service.machine),
```

Comment:
I think we should only run the serializer over "user values" - the vast majority of `service.machine` itself is needed untouched for us to work with the inspector etc. 

I see how this solves the problem of serializing context embedded in the machine, parametrized actions, and stuff like this but it feels a little bit iffy to just run the custom serializer over the whole thing. 

What do you think? Either way - I'm OK with this right now but would prefer to add a TODO comment or something here for the future, unless you think that my concerns are not valid.

Reply:
There isn't an easy way to do this if we want to target parameterized actions, context, etc. Let's revisit this if it becomes a problem.